lib/silencer/rails/logger.rb

```diff
@@ -45,7 +45,7 @@ def call(env)
       private
 
       def quiet(&block)
-        if ::Rails.logger.respond_to?(:silence)
+        if ::Rails.logger.respond_to?(:silence) && ::Rails.logger.method(:silence).owner != ::Kernel
```

Comment:
Can you explain the rationale around `method(:silence).owner != ::Kernel`. 

I've been under the impression that `Rails.logger.silence` was a Rails-only implementation, what does the `owner != ::Kernel` safeguard against?